cmd/manager/aggregator.go

```diff
@@ -571,6 +585,10 @@ func createResults(crClient aggregatorCrClient, scan *compv1alpha1.ComplianceSca
 		if err := createOrUpdateOneResult(crClient, scan, checkResultLabels, checkResultAnnotations, checkResultExists, pr.CheckResult); err != nil {
 			return fmt.Errorf("cannot create or update checkResult %s: %v", pr.CheckResult.Name, err)
 		}
+		// Forward pr.CheckResult to an external endpoint, if
+		// configured to do so, here.
+		// c := pr.CheckResult
+		// f.SendComplianceCheckResult(c)
```

Comment:
I am wondering what we do here on things like updating a CCR object?

Reply:
The "consumer" (regardless of local or network) should do the work, because only the consumer knows how are the results stored (a CR in etcd? A record in a DB?)

Reply:
Yes - exactly. That would be up to the compliance service to rationalize.

Are you thinking the compliance operator would conditionally send objects based on it if was an update?

Reply:
Not really, but I just realized that the "consumer" might also need to handle at least object deletes.
Consider you run a scan and get three results, A, B, C
Then you upgrade to a new content version which removes rule A. Re-run the scan, now the aggregator only sends B, C.

That could be handled on the consumer side, e.g. by having a UUID or a timestamp associated with the latest results and anything that was not touched would be deleted, but it's something to keep in mind.

Reply:
I don't think we can assume a kube-style CRUD on the consumer side. If we want the client to be fire-and-forget, then the server would receive the serialized results in full every time, and it can compare its previous state of the results with the new payload. We could probablye use a token ID or client cert name as the identifier.

Reply:
I was thinking the compliance-operator/aggregator would use a fire-and-forget method.

Let's assume the compliance-operator runs a scan that generates three results, `foo`, `bar`, and `bazz`. It puts all those on the wire to the compliance service. The compliance service would timestamp those results, and persist them accordingly.

Say the `bazz` result get deleted, or becomes obsolete such that it's excluded from a subsequent scan. When the scan is re-run, the aggregator puts the `foo` and `bar` result on the wire to the compliance service. The compliance service would timestamp those results and persist them. In my mind, this is separate from the first scan, since it's conducted at a later point in time. The compliance service wouldn't go back and fish the `bazz` result from the first scan and delete it (if that's what we were talking about).

Instead, the compliance service would have a data set that would support trending over time. A consumer of the compliance service could see that `bazz` was run in scan `88ed908d-bb05-4bd9-9a00-64da9be4fc49` which happened on Monday morning, but it didn't run in scan `ae22c717-6ba5-410b-8195-2781ef04d3d4`, which happened on Tuesday afternoon.

Reply:
I see, so ultimately the client should not care about the delta and neither should the service.

Reply:
> I see, so ultimately the client should not care about the delta and neither should the service.

I don't think the client should, or can. The compliance-operator doesn't guarantee storage of results up to a certain time window, right? If my understanding is correct, it does not, which makes it difficult for the compliance operator make decisions about what to send based on previous, or future, scans.

In that case, I think it would be simpler to forward everything, and deduce patterns about the data set in the compliance service.

I can't think of a use-case where the compliance service would go back and delete a past scan result based on a new result that just came in. So, I think you're right in both cases.

In the off chance we do identify a use-case where the compliance service should correct a scan result, then we only have to do that in a single place, the compliance service.